framework/Source/Operations/ScreenBlend.metal

```diff
@@ -2,15 +2,112 @@
+                                   texture2d<half> inputTexture2 [[texture(1)]],
+                                   constant ScreenBlendUniform& uniform [[ buffer(1) ]])
 {
     constexpr sampler quadSampler;
     half4 textureColor = inputTexture.sample(quadSampler, fragmentInput.textureCoordinate);
-    constexpr sampler quadSampler2;
-    half4 textureColor2 = inputTexture2.sample(quadSampler, fragmentInput.textureCoordinate2);
+
+    // Calculate the aspect ratios of the two textures
+    float dimension1 = max(inputTexture.get_width() , inputTexture.get_height());
+    float dimension2 = max(inputTexture2.get_width() , inputTexture2.get_height());
+
+    float2 norSize1 = float2(inputTexture.get_width() / dimension1, inputTexture.get_height() / dimension1);
+    float2 norSize2 = float2(inputTexture2.get_width() / dimension2, inputTexture2.get_height() / dimension2);
+
+    float scaleX = norSize1.x / norSize2.x;
+    float scaleY = norSize1.y / norSize2.y;
+
+    float2 textureCoordinate2 = fragmentInput.textureCoordinate; // default scale fill
+
+    // Linear Interpolation
+    // f(min) = a
+    // f(max) = b
+    // f(x) = ((b - a)(x - min) / (max - min)) + a
+    if (uniform.mode == 0.0) { 
+        // aspect fill
+        float scale = max(scaleX, scaleY);
+        if (scaleX > scaleY) {
+            // fill width
+            float offsetY = abs(norSize1.y - norSize2.y * scale) / 2;
+            offsetY /= scale;
+            float min = 0;
+            float max = 1;
+            float a = offsetY;
+            float b = 1 - offsetY;
+            textureCoordinate2.y = (b - a) * (textureCoordinate2.y - min) / (max - min) + a;
+        } else {
+            // fill height
+            float offsetX = abs(norSize1.x - norSize2.x * scale) / 2;
+            offsetX /= scale;
+            float min = 0;
+            float max = 1;
+            float a = offsetX;
+            float b = 1 - offsetX;
+            textureCoordinate2.x = (b - a) * (textureCoordinate2.x - min) / (max - min) + a;
+        }
+    } else if (uniform.mode == 1.0) { 
+        // aspect fit
+        float scale = min(scaleX, scaleY);
+        if (scaleX < scaleY) {
+            // fit width
+            float offsetY = (norSize1.y - norSize2.y * scale) / 2;
+            float min = offsetY;
+            float max = 1 - offsetY;
+            float a = 0;
+            float b = 1;
+            textureCoordinate2.y = (b - a) * (textureCoordinate2.y - min) / (max - min) + a;
+        } else {
+            // fit height
+            float offsetX = (norSize1.x - norSize2.x * scale) / 2;
+            float min = offsetX;
+            float max = 1 - offsetX;
+            float a = 0;
+            float b = 1;
+            textureCoordinate2.x = (b - a) * (textureCoordinate2.x - min) / (max - min) + a;
+        }
+    } else {
+        // scale fill
+        textureCoordinate2 = fragmentInput.textureCoordinate;
+    }
```

Comment:
It is clear. But I think there is a more simple calculation.
```
if (uniform.mode == 0.0) { 
// aspect fill
float scale = max(scaleX, scaleY);
textureCoord2 = (norSize2 * scale) * (fragmentInput.textureCoordinate - 0.5) + 0.5;
} else if (uniform.mode == 1.0) { 
float scale = min(scaleX, scaleY);
textureCoord2 = (norSize2 * scale) * (fragmentInput.textureCoordinate - 0.5) + 0.5;
} else {
textureCoord2 = fragmentInput.textureCoordinate
}

```

Reply:
@Justin-Ruan 

This didn't work for me. That's why I was asking about texture coordinates yesterday.

The aspect fit result look like this which is not expected.

![image](https://github.com/cardinalblue/GPUImage3/assets/122647/bead5121-30ca-4352-83bd-6cdfc27c97c3)

Reply:
@yyjim Please try this one
```
// Calculate the aspect ratios of the two textures
float2 size1 = float2(inputTexture.get_width() , inputTexture.get_height());
float2 size2 = float2(inputTexture2.get_width() , inputTexture2.get_height());

float ratio1 = size1.x / size1.y;
float ratio2 = size2.x / size2.y;

float2 scale = 1.0;

if (mode == 0) {
    // Fill
    float rate = min(ratio1, ratio2);
    scale = float2(ratio1, ratio2) / rate;
} else if (mode == 1) {
    // Fit
    float rate = max(ratio1, ratio2);
    scale = float2(ratio1, ratio2) / rate;
}

float2 textureCoordinate2 = (textureCoordinate - 0.5) * scale + 0.5;
```

Reply:
@Justin-Ruan 

@Vivienne0206's simplified version is correct 🥳
The formula is much cleaner.

Thanks for the suggestion.

Reply:
Great! 👍